aiogram/dispatcher/filters/builtin.py

```diff
@@ -691,3 +692,24 @@ def __init__(self, is_forwarded: bool):
 
     async def check(self, message: Message):
         return bool(getattr(message, "forward_date")) is self.is_forwarded
+
+
+class ChatTypeFilter(BoundFilter):
+    key = 'chat_type'
+
+    def __init__(self, chat_type: typing.Container[ChatType]):
+        if isinstance(chat_type, str):
+            chat_type = {chat_type}
+
+        self.chat_type: typing.Set[str] = set(chat_type)
+
+    async def check(self, obj: Union[Message, CallbackQuery]):
+        if isinstance(obj, Message):
+            obj = obj.chat
+        elif isinstance(obj, CallbackQuery):
+            obj = obj.message.chat
+        else:
+            logger.warning("ChatTypeFilter doesn't support %s as input", type(obj))
```

Comment:
Is better to use `warnings.warn` here

Reply:
I'd do `assert isinstance(obj, (Message, CallbackQuery))` before `if/elif` and in else `return False`

Reply:
I don't really see any point in breaking the whole pipeline of filters. As far as I remember, that's the reason why we didn't use `raise NotImplementedError`. I'd like to keep the warning message in such case. Remember, that in other filters we don't have such way to handle unexpected input

Reply:
fixed in https://github.com/aiogram/aiogram/pull/356/commits/0eb5a476fae8f28ed286db5e2df844aa31f57b14